### PR TITLE
BY 3.0 ported licenses

### PR DESCRIPTION
**Pull request by dpoirier**
## Description
Adds the BY 3.0 ported licenses to the site, by just saving the license part of the HTML from their existing HTML pages and embedding that in a page with the new look.  Adds a nicer test/dev page with links to all the deeds and licenses at /licenses/all/.

## Checklist
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: cc_licenses/templates/includes/legalcode_30_unported_license.html [hunk 1]
```diff
@@ -5,7 +5,7 @@
 <div id="legal-code-body" class="padding-larger margin-top-bigger has-text-black">
   {% include 'includes/snippet/icon_header_snippet.html' %}
   <div>
-    <h3 class="padding-bottom-normal b-header">Creative Commons {{ license.title_english }}</h3>
```
Comment (Audiosutras) [inline cc_licenses/templates/includes/legalcode_30_unported_license.html:8]: It looks like we are missing the `title` in the license header, side menu, and legal code section.

Comment (dpoirier) [reply cc_licenses/templates/includes/legalcode_30_unported_license.html:8]: I think running migrations and load_html_files again will resolve this, but let me know if not.

File: cc_licenses/templates/all_licenses.html [hunk 2]
```diff
@@ -0,0 +1,53 @@
+{% extends "base.html" %}
```
Comment (Audiosutras) [inline cc_licenses/templates/all_licenses.html:1]: Legal Code section for unported licenses are missing a closing `div`. Sections that are meant to be distinct have the legal code section styles applying to them (they are included in the legal code section).

**Review by Audiosutras — COMMENTED** [review]
We need to update the side menu.

**Review by Audiosutras — COMMENTED** [review]
For these licenses, can we format/style the legal code to be more inline with the legal code sections of the other licenses? Is this feasible?

File: cc_licenses/templates/all_licenses.html (hunk 2 above)
Comment (dpoirier) [reply cc_licenses/templates/all_licenses.html:1]: @Audiosutras can you expand on this? The template being commented on has no `div` tags to close.

Comment (Audiosutras) [reply cc_licenses/templates/all_licenses.html:1]: @dpoirier 

image

Comment (Audiosutras) [reply cc_licenses/templates/all_licenses.html:1]: Starting at `View Legal Code as plain text` and down, these sections should be outside of the legal code container.

**Comment by dpoirier** [comment]
> We need to update the side menu.

Right now there's not a good way to do a side menu for the ported licenses. The ported licenses vary in which sections they include, and how they're identified in the HTML. They might not even all have link targets to link to.

**Comment by dpoirier** [comment]
> For these licenses, can we format/style the legal code to be more inline with the legal code sections of the other licenses? Is this feasible?

I'll take a look.

File: cc_licenses/templates/all_licenses.html (hunk 2 above)
Comment (Audiosutras) [reply cc_licenses/templates/all_licenses.html:1]: Sorry about the inaccurate comment location.

**Comment by Audiosutras** [comment]
> > We need to update the side menu.
> 
> Right now there's not a good way to do a side menu for the ported licenses. The ported licenses vary in which sections they include, and how they're identified in the HTML. They might not even all have link targets to link to.

I understand.

**Comment by dpoirier** [comment]
> > For these licenses, can we format/style the legal code to be more inline with the legal code sections of the other licenses? Is this feasible?
> 
> I'll take a look.

I can't think of a good way to do this. We're just embedding the license part of the HTML from the current HTML files unmodified, so they don't have vocabulary classes on them. I guess we could copy the style rules that end up being applied to parts of the other license pages, and try to put them in a style file with selectors based on the structure of the HTML, but I don't think we really want to head down that road.

File: cc_licenses/templates/all_licenses.html (hunk 2 above)
Comment (dpoirier) [reply cc_licenses/templates/all_licenses.html:1]: It looks like there's a missing `</div>` in `legalcode_30_ported_license.html`, I'll fix that.

**Comment by Audiosutras** [comment]
> > > For these licenses, can we format/style the legal code to be more inline with the legal code sections of the other licenses? Is this feasible?
> > 
> > 
> > I'll take a look.
> 
> I can't think of a good way to do this. We're just embedding the license part of the HTML from the current HTML files unmodified, so they don't have vocabulary classes on them. I guess we could copy the style rules that end up being applied to parts of the other license pages, and try to put them in a style file with selectors based on the structure of the HTML, but I don't think we really want to head down that road.

Got it. I don't think so either.

File: cc_licenses/templates/all_licenses.html (hunk 2 above)
Comment (Audiosutras) [reply cc_licenses/templates/all_licenses.html:1]: looks much better.

**Review by Audiosutras — APPROVED** [review]
💯 Looks good Dan!

**Pull request merged**